### PR TITLE
Torch-based GEMM with fake dequantization for mxfp6 e2m3

### PR DESCRIPTION
**Pull request by nirmie**
- Packs f32 into fp6 and unpacks back to f32 for matmul
- Pytests for quantization error, matmul error, and LUT rounding
- Does not use CDNA4 FP6, only fake dequantization

Will soon work on adding CDNA4 native mxfp6 operations

### CODE REVIEW

File: tests/kernel/fake_dequant_mxfp6.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    """
+    Packs FP32 tensor into MXFP6 E2M3 format.
+    x: [N, K]
+    Returns: packed_uint8 [N, (K*6)/8], scales [N, K/block_size]
+    """
+
+    # Calculate block-wise scales (Microscaling)
+    x_reshaped = x.view(-1, block_size)
+    amax = x_reshaped.abs().max(dim=1, keepdim=True).values
+    scales = amax / MAX_E2M3
```
[the remark this answers is not in the document]
Comment (Giuseppe5) [reply tests/kernel/fake_dequant_mxfp6.py]: Changing line 49/50 to (assuming MAX_E2M3 is also a torch.tensor:

```python
amax = torch.pow(2, torch.ceil(torch.log2(x_reshaped.abs().max(dim=1, keepdim=True).values)))
scales = amax /  torch.pow(2, torch.ceil(torch.log2(MAX_E2M3)))
```

Will fix the computation of the scale, however it seems that there is still a mismatch in the quant tensor itself due to potential normalization issues
